framework/include/auxkernels/ElementL2IntegralAux.h

```diff
@@ -0,0 +1,52 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef ELEMENTL2INTEGRALAUX_H
+#define ELEMENTL2INTEGRALAUX_H
+
+// MOOSE includes
+#include "FunctionAux.h"
+
+// Forward declarations
+class ElementL2IntegralAux;
+
+template<>
+InputParameters validParams<ElementL2IntegralAux>();
+
+/**
+ * Compute an elemental field variable which is a single value per
+ * element equal to the L2-norm of a given function which is defined
+ * by derived classes.  The user is responsible for overriding the
+ * computeValue() function to return the quantity which should be
+ * squared, this class takes care of squaring it, doing the
+ * integration, and finally taking the square root.
+ */
+class ElementL2IntegralAux : public AuxKernel
```

Comment:
Ok - it's totally my fault... but should this class be named `ElementL2NormAux`?  An "L2 Integral" is not a thing (although I'm sure most people can understand what it means).

I'm really sorry about that... I just said the wrong thing.